core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -183,6 +185,11 @@ class KafkaApis(val requestChannel: RequestChannel,
     val updateMetadataResponse =
       if (authorize(request.session, ClusterAction, Resource.ClusterResource)) {
         replicaManager.maybeUpdateMetadataCache(correlationId, updateMetadataRequest, metadataCache)
+        if (controller.isActive && !updateMetadataRequest.partitionStates.isEmpty) {
```

Comment:
On second thought, should we check controller.isActive here? If the broker switched to non-controller after taking the createTopicRequest, doing this check will force the request to wait for the timeout, which is probably not necessary.

There is probably also no need to check partitionStates.isEmpty since foreach takes care of that naturally.